### ISSUE TITLE
Terraform panic: interface conversion: interface {} is []interface {}, not string

### ISSUE THREAD

**Issue by SarahRiley** [terraform-providers/terraform-provider-aws#10528]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.12.10

### Affected Resource(s)

* plugin.terraform-provider-aws_v2.32.0_x4
* aws_config_config_rule

We also see this in:

* aws_cloudwatch_event_target

(i can go back and get this trace if you need it)

* An output with a value of aws_s3_bucket.cloudtrail.id:

```
2019/10/16 20:53:46 [TRACE] EvalWriteOutput: Saving Create change for module.logs_prod.output.cloudtrail_bucket_name in changeset
2019-10-16T20:53:46.758Z [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: panic: interface conversion: interface {} is []interface {}, not string
2019-10-16T20:53:46.758Z [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4:
2019-10-16T20:53:46.758Z [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: goroutine 158 [running]:
2019-10-16T20:53:46.759Z [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: github.com/jen20/awspolicyequivalence.newAWSStringSet(0x3c8d1e0, 0xc0009e2820, 0xc0000c6dc0
, 0xb, 0xb)
2019-10-16T20:53:46.759Z [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/jen20/awspolicyequivalenc
e@v1.0.0/aws_policy_equivalence.go:386 +0x212
2019-10-16T20:53:46.759Z [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4: github.com/jen20/awspolicyequivalence.(*awsPolicyStatement).equals(0xc0003b86c0, 0xc0003b89
90, 0x0)
2019-10-16T20:53:46.759Z [DEBUG] plugin.terraform-provider-aws_v2.32.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/jen20/awspolicyequivalenc
e@v1.0.0/aws_policy_equivalence.go:176 +0x2d8
2019-10-16T20:53:46.758Z [DEBUG] plugin: plugin process exited: path=/workdir/terraform/.terraform/plugins/linux_amd64/terraform-provider-aws_v2.32.0_x4 pid=500
```

### Terraform Configuration Files

```hcl
resource "aws_config_config_rule" "encrypted_volumes" {
  name        = "encrypted-volumes"
  description = "Checks whether EBS volumes that are in an attached state are encrypted."

  source {
    owner             = "AWS"
    source_identifier = "ENCRYPTED_VOLUMES"
  }

  depends_on = [aws_config_configuration_recorder.config_recorder]
}
```

### Panic Output

https://gist.github.com/SarahRiley/9cf34603de6daea93ce20bc13a703c66

### Expected Behavior

A plan

### Actual Behavior

A crash

### Steps to Reproduce

1. `terraform apply`

**Comment by ewbankkit** [comment]
@SarahRiley Can you please paste an example of the IAM policy (sanitized where necessary of cource) on the config recorder that is causing this panic.
I see the [line of code](https://github.com/jen20/awspolicyequivalence/blob/9ebbf3c225b2b9da629263e13c3015a5de7965d1/aws_policy_equivalence.go#L386) where the panic is raised but would like to get an understanding of the structure of the IAM policy.
Thanks.

**Comment by SarahRiley** [comment]
Certainly, here is the relevant code:

```
resource "aws_config_configuration_recorder" "config_recorder" {
  name     = "Configuration-Recorder"
  role_arn = aws_iam_role.aws_config.arn
}

resource "aws_iam_role" "aws_config" {
  name = "aws-config"

  assume_role_policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "config.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
POLICY

}

resource "aws_iam_role_policy" "aws_config" {
  name = "aws-config"
  role = aws_iam_role.aws_config.id

  policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "s3:*"
      ],
      "Effect": "Allow",
      "Resource": [
        "${aws_s3_bucket.aws_config.arn}",
        "${aws_s3_bucket.aws_config.arn}/*"
      ]
    }
  ]
}
POLICY

}

resource "aws_iam_role_policy_attachment" "aws_config" {
  role       = aws_iam_role.aws_config.name
  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSConfigRole"
}

# S3

resource "aws_s3_bucket" "aws_config" {
  bucket = "REDACTED"
  acl    = "private"
  region = var.aws_region
}
```

**Comment by ewbankkit** [comment]
Similar:
* https://github.com/terraform-providers/terraform-provider-aws/issues/10135

**Comment by ewbankkit** [comment]
Submitted upstream PR https://github.com/jen20/awspolicyequivalence/pull/9 to fix the panic, but this won't fix the underlying issue and will probably result in a run time `MalformedPolicyDocument: Syntax errors in policy.` error.

**Comment by SarahRiley** [comment]
Interesting. Where’s the problem in the policy, if you don’t mind?  This was upgraded from 0.11 code originally, where it was valid.

**Comment by SarahRiley** [comment]
OK, I've narrowed down the problem to this:

```
resource "aws_iam_role_policy" "vulnerability_scanning_iam_policy" {
  count = var.fargate_enabled ? 1 : 0

  name = "vulnerability-scanning-iam-policy"

  role = module.elastic_spyglass_fargate.iam_role_id

  policy = <<EOF
{
	"Version": "2012-10-17",
	"Statement": [{
			"Effect": "Allow",
			"Action": "sts:AssumeRole",
			"Resource": ${jsonencode(local.scanned_account_id_list)}
		},
...
EOF

locals {
  scanned_account_id_list = [formatlist(
    "arn:aws:iam::%s:role/${module.elastic_spyglass_fargate.iam_role_id}",
    var.scanned_account_ids,
  )]
}

variable "scanned_account_ids" {
  type = list(string)

  default = []
}
}
```

Removing the outer []s for local.scanned_account_id_list fixed it.

**Comment by ewbankkit** [comment]
@SarahRiley Thanks for diving into this. The list-in-a-list was the same error as in https://github.com/terraform-providers/terraform-provider-aws/issues/10135 but we should still get the the PR to stop the panic merged.
Could you please close this issue?

**Comment by SarahRiley** [comment]
Agreed. Thanks!

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.42.0 of the Terraform AWS Provider, likely next week. 👍
